src/Illuminate/Support/Facades/Cache.php

```diff
@@ -49,6 +49,7 @@
  * @method static string getPrefix()
  * @method static \Illuminate\Contracts\Cache\Lock lock(string $name, int $seconds = 0, string|null $owner = null)
  * @method static \Illuminate\Contracts\Cache\Lock restoreLock(string $name, string $owner)
+ * @method static bool isLocked(string $name)
```

Comment:
As our facade docblocks are auto generated, I would recommend putting an `isLocked` method on the Cache Respository. It could throw an exception if the store doesn't implement `LockProvider`. This should mean that it will appear in the Facade docblock.

Reply:
Would you also recommend adding the `isLocked` method to the `LockProvider`? Originally I avoided doing this for BC, but could re-target to 10.x if that is desired.